src/components/Posts/NewPost.jsx

```diff
@@ -0,0 +1,54 @@
+import { UserContext } from "../UserContext";
+import { useContext, useState } from "react";
+
+import {
+  NewPostCont,
+  NewPostHeader,
+  InputName,
+  TitleInputContainer,
+  BodyContainer,
+  NewPostBtn,
+} from "./NewPost.style";
+
+const NewPost = (props) => {
+  const [title, setTitle] = useState("");
+  const [body, setBody] = useState("");
+  const { user } = useContext(UserContext);
+  const openPost = props.openPost;
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    await fetch("https://jsonplaceholder.typicode.com/", {
+      method: "POST",
+      body: JSON.stringify({
+        title,
+        body,
+        userId: user.id,
+      }),
+      headers: {
+        "Content-type": "application/json; charset=UTF-8",
+      },
+    });
```

Comment:
handle all your API requests, show the user an error if it exists, if not then okay, continue the flow